### ISSUE TITLE
class-canvas.php filters the_content() & not get_the_content()

### ISSUE THREAD

**Issue by toderash** [andrew-worsfold/tailor#81]
This vexed me for a while, given the error message "The canvas does not exist in the page.  This could be caused by a plugin or theme conflict." is really not all that helpful. After chasing down all the things, I eventually found that my content section didn't have
`'<div id="canvas"></div>' .
 '<div class="tools" id="tools">' .
 '<div class="guide-container" id="guide"></div>' .
 '<div class="select-container" id="select"></div>' .
 '</div>';`
as added by `canvas_content( $content )` in `class-canvas.php`. Turns out this is because in my theme, I'm echoing get_the_content() rather than using the_content(). The latter is filtered (line 57 of `class-canvas.php`) to return the code above, but not the former.

Easy to change in the theme, but this could cause issues for others. Was it a conscious decision not to filter `get_the_content()` or should this change be made now? If this is intentional for some reason, a quick mention in a FAQ or Themer's Guide would be very helpful for others who may be troubleshooting.

**Comment by Arood** [comment]
If I remember correctly, get_the_content() doesn't apply the `the_content` filter that automatically gets applied to the_content(), and perhaps that's why it doesn't work for you. Does it work if you apply that filter manually? `$content = apply_filters('the_content', $content);`

**Comment by toderash** [comment]
Yes, that's correct, so this would be a bug in the theme, which should use `apply_filters()` manually if using `get_the_content()`. I'm sure I've seen attempts to use `add_filter()` with `'get_the_content'`, but clearly it was wrong wherever I saw it. I just checked, and there's no filter applied anywhere in the `get_the_content()` function other than the 'more' link.

Anyway, documented here as a troubleshooting step in case anyone finds a theme that barfs on attempts to Tailor the page.
